src/big/big_float.cr

```diff
@@ -51,16 +83,36 @@ struct BigFloat < Float
     LibGMP.mpf_cmp(self, other)
   end
 
-  def <=>(other : Float)
+  def <=>(other : BigInt)
+    LibGMP.mpf_cmp_z(self, other)
+  end
+
+  def <=>(other : Float32 | Float64)
     LibGMP.mpf_cmp_d(self, other.to_f64)
   end
 
   def <=>(other : Int::Signed)
-    LibGMP.mpf_cmp_si(self, other.to_i64)
+    if LibGMP::Long == Int64
```

Comment:
I'd prefer this method to be split up into this:
```cr
def <=>(other : Int::Signed)
  LibGMP.mpf_cmp_si(self, LibGMP::Long.new(other))
end

{% unless LibGMP::Long == Int64 %}
  def <=>(other : Int64)
    # On 32bit platforms, we have to go via float for 64bit integers.
    LibGMP.mpf_cmp(self, other.to_big_f)
  end
{% end %}
```

And use the same technique on any other applicable methods. For example you seem to be using a different overloading strategy for `def /` than `def <=>`.